src/poly/mod.rs

```diff
@@ -12,3 +12,74 @@ pub use element::PolyElem;
 pub use matrix::PolyMatrix;
 pub use params::PolyParams;
 pub use polynomial::Poly;
+
+use sampler::{DistType, PolyUniformSampler};
+use std::sync::Arc;
+
+pub fn rlwe_encrypt<M, SU>(
```

Comment:
@enricobottazzi Can you define this function and unit tests not in mod.rs but in enc.rs under poly?